pkg/builder/cluster/convert/convert.go

```diff
@@ -105,48 +105,76 @@ var (
 )
 
 // TODO(mattmoor): Should we move this somewhere common, because of the flag?
-func gitToContainer(git *v1alpha1.GitSourceSpec) (*corev1.Container, error) {
+func gitToContainer(git *v1alpha1.GitSourceSpec, name string) (*corev1.Container, error) {
 	if git.Url == "" {
 		return nil, validation.NewError("MissingUrl", "git sources are expected to specify a Url, got: %v", git)
 	}
 	if git.Revision == "" {
 		return nil, validation.NewError("MissingRevision", "git sources are expected to specify a Revision, got: %v", git)
 	}
+
+	args := []string{"-url", git.Url,
+		"-revision", git.Revision,
+	}
+	containerName := initContainerPrefix + gitSource
+
+	if name != "" {
+		// update container name to suffix source name
+		args = append(args, []string{"-name", name}...)
+		containerName = containerName + "-" + name
+	}
+
 	return &corev1.Container{
-		Name:  initContainerPrefix + gitSource,
-		Image: *gitImage,
-		Args: []string{
-			"-url", git.Url,
-			"-revision", git.Revision,
-		},
+		Name:         containerName,
+		Image:        *gitImage,
+		Args:         args,
 		VolumeMounts: implicitVolumeMounts,
 		WorkingDir:   workspaceDir,
```

Comment:
If every source container uses the same working dir will multiple sources clobber each other?

Reply:
Nevermind! the answer here is clearly no :)